### ISSUE TITLE
`nvim_call_function` with non-string arguments

### ISSUE THREAD

**Issue by HiPhish** [esensar/neovim-java#96]
The `NeovimAPI.callFunction` method takes a `List<String>` object as its second parameter. The list elements are passed as string arguments to the Neovim function, but this results in an error if the function expects anything other than a string. That makes the `callFunction` method pretty much useless.

As an example consider the following code:

```java
var projectDirectory = (String) api.callFunction("getcwd", List.of("0", "0")).get();
```

This will throw an `ExecutionException` with the message `java.util.concurrent.ExecutionException: RPCException{rpcError=RPCError{id=0, message='Vim:E474: Invalid argument'}}`. That is the same message I get when I execute `echo nvim_call_function('getcwd', ['0', '0'])` (note the quotation marks).

**Comment by esensar** [comment]
Thanks for noticing this, it is a simple fix, since serializer is already ready for any argument type, so it is a simple change in the interface.

**Comment by esensar** [comment]
Fix released in 0.1.14

**Comment by HiPhish** [comment]
Now I am getting

```
/home/hiphish/Developer/vim/gradle.nvim/src/main/java/org/gradle/tooling/nvim/StdIoRpcConnection.java:7: error: cannot access RPCConnection
import com.ensarsarajcic.neovim.java.corerpc.client.RPCConnection;
                                                   ^
  bad class file: /home/hiphish/.gradle/caches/modules-2/files-2.1/com.ensarsarajcic.neovim.java/core-rpc/0.1.14/c0f6861e3c8b1965a14280319a335e561858a475/core-rpc-0.1.14.jar(/com/ensarsarajcic/neovim/java/corerpc/client/RPCConnection.class)
    class file has wrong version 57.0, should be 55.0
    Please remove or make sure it appears in the correct subdirectory of the classpath.
/home/hiphish/Developer/vim/gradle.nvim/src/main/java/org/gradle/tooling/nvim/StdIoRpcConnection.java:17: error: cannot find symbol
public final class StdIoRpcConnection implements RPCConnection {
                                                 ^
  symbol: class RPCConnection
/home/hiphish/Developer/vim/gradle.nvim/src/main/java/org/gradle/tooling/nvim/GradleRpcListener.java:5: error: cannot access RPCListener
import com.ensarsarajcic.neovim.java.corerpc.client.RPCListener;
```

and so on, when I want to compile the code. I bumped the version number to `0.1.14` in my `build.gradle`. Should I have done something more, or do I need to wait for the package repository to catch up?

**Comment by esensar** [comment]
0.1.14 also bumps up to JDK 13, so that is probably the issue here.

**Comment by HiPhish** [comment]
Is it not possible to use the library with Java 131? I would prefer staying with the latest LTS release because it makes it easier for people to install the plugin. The next planned LTS release is Java 17, scheduled for September 2021.

**Comment by esensar** [comment]
That makes sense. I will have to check it out, it should be compatible with it.

**Comment by esensar** [comment]
Thanks for the comment once again. I have released 0.1.15 with JDK 11 support and it will stay on that version until the next LTS release.
